contributors/design-proposals/scheduling/coscheduling.md

```diff
@@ -0,0 +1,180 @@
+---
+kep-number: 24
+title: Coscheduling
+authors:
+  - "@k82cn"
+owning-sig: sig-scheduling, machine-learning WG
+reviewers:
+  - "@bsalamat"
+  - "@vishh"
+approvers:
+  - "@bsalamat"
+  - "@vishh"
+editor: TBD
+creation-date: 2018-07-03
+last-updated: 2018-09-19
+status: provisional
+---
+
+# Coscheduling
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Function Detail](#function%20detail)
+  * [API Definitation](#api%20definition)
+  * [Lifecycle Management](#lifecycle%20management)
+  * [kube-arbitrator](#kube-arbitrator)
+  * [Customized Controller](#customized%20controller)
+* [Feature Interaction](#feature%20interaction)
+  * [Multi-scheduler](#multi-scheduler)
+  * [Priority/Preemption](#priority/preemption)
+  * [Pod RestartPolicy](#pod%20restartPolicy)
+  * [Admission Controller](#admission%20controller)
+  * [Kubectl](#kubectl)
+* [References](#references)
+
+## Motivation
+
+Kubernetes has become a popular solution for orchestrating containerized workloads; it has been largely successful in orchestrating serving and storage workloads, and, with native K8s support for Spark. Meanwhile, the community also try to run Machine Learning (ML) workloads on Kubernetes, e.g. [kubeflow/tf-operator](https://github.com/kubeflow/tf-operator). When running a Tensorflow/MPI job, all tasks of a job must be start together; otherwise, did not start anyone of tasks. If the resource is enough to run all 'tasks', everything is fine; but it's not true for most of case, especially in the on-prem environment. In worst case, all jobs are pending here because of deadlock: every job only start part of tasks, and waits for the other tasks to start. It'll be worse in federation for cross-domain case which is out of scope in this doc.
+
+After the discussion at [Coscheduling/Gang-scheduling](https://docs.google.com/document/d/1AUwcvTtULNvow5M9e428FnlvINO1uQ7ojRoTGuTp4DA/edit#heading=h.ckn8nv2jj0xv) proposal, we decide to implement Coscheduling in [kube-arbitrator](https://github.com/kubernetes-incubator/kube-arbitrator) by CRDs. kube-arbitrator focuses on "batch" workload in kubernetes, and will share the same [scheduling frameworks](https://github.com/kubernetes/community/pull/2281) when it's ready. This document is used to provide definition of API object and the scheduler behaviour of Coscheduling.
+
+## Function Detail
+
+### API Definition
+
+The following requirements are identified during the discussion of this feature:
+
+1. Existing workload can use this feature without (or with a few) configuration changes
+2. Pods of a group/gang may have different `PodSpec` (and/or belong to different collections)
+3. Existing controllers which are responsible for managing life cycle of collections work well with this feature
+
+To meet the requirements above, the following **Kind** is introduced by CRD under `scheduling.incubator.k8s.io/v1alpha1` **Group**/**Version**.
```

Comment:
nit: Should it be `incubator.scheduling.k8s.io`? That way once it's out of incubation we only need to drop a sub-domain?

Reply:
Yours seems better :)